sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -123,60 +189,24 @@ class EthService(val stateView: AccountStateView, val nodeView: CurrentView[Acco
-    Numeric.toHexString(getFrom(tx).pubKeyBytes), Numeric.toHexString(ethTx.getData), Numeric.toHexString(ethTx.getSignature.getR), Numeric.toHexString(ethTx.getSignature.getS), Numeric.toBigInt(getEthereumTransaction(tx).getSignature.getV).longValue(), String.valueOf(getEthereumTransaction(tx).version), Numeric.toHexStringWithPrefix(ethTx.getMaxFeePerGas), Numeric.toHexStringWithPrefix(ethTx.getMaxPriorityFeePerGas), new util.ArrayList[Any]()) // TODO: access list)
-
   @RpcMethod("eth_getTransactionReceipt")
-  def getTransactionReceipt(transactionHash: String): String = { // TODO: Receipts will be supported later on
-    return "{\"transactionHash\": " + transactionHash + "," +
-      "\"transactionIndex\":  \"0x1\"," +
-      "\"blockNumber\": \"0xb\"," +
-      "\"blockHash\": \"0xc6ef2fc5426d6ad6fd9e2a26abeab0aa2411b7ab17f30a99d3cb96aed1d1055b\"," +
-      "\"cumulativeGasUsed\": \"0x33bc\"," +
-      "\"gasUsed\": \"0x4dc\"," +
-      "\"contractAddress\": \"0xb60e8dd61c5d32be8058bb8eb970870f07233155\"," +
-      "\"logs\": []," +
-      "\"logsBloom\": \"0x0\"," +
-      "\"status\": " + 0x1 + "}"
-    /*new TransactionReceipt(transactionHash, "", // TODO: get transaction index
-      "", // get block hash
-      "", // get block number
-      "", // get cumulative gas used
-      "", // get gas used
-      "", // return contract address if one is created
-      "", // return root
-      "1", // return status
-      "", // add from
-      "", // add to
-      new util.ArrayList[Log](),// return logs
-      "", // return logsBloom
-      "", // insert revert reason
-      "", // insert tx type
-      "" // insert effective gas price*/
-    //)
-  }
-
-  @RpcMethod("eth_getCode") def getCode(address: String, tag: Quantity): String = { // TODO: return bytecode of given address or empty
-    ""
-  }
-
-  private def getBalance(address: String, tag: Quantity) = { // TODO: Add blockNumberOrTag handling
-    val balance = nodeView.state.getBalance(Numeric.hexStringToByteArray(address))
-    Numeric.toHexStringWithPrefix(balance)
+  def getTransactionReceipt(transactionHash: String): EthereumReceiptView = {
+    val receipt = stateView.getTransactionReceipt(Numeric.hexStringToByteArray(Numeric.cleanHexPrefix(transactionHash)))
+    val tx = nodeView.history.searchTransactionInsideBlockchain(Numeric.cleanHexPrefix(transactionHash))
+    if (receipt.isEmpty || tx.isEmpty) {
+      new EthereumReceiptView()
+    } else {
+      new EthereumReceiptView(receipt.get, tx.get().asInstanceOf[EthereumTransaction])
```

Comment:
I think, we don't need to pass `tx` to `EthereumReceiptView`. All we need is stored in the `EthereumReceipt`

Reply:
No, the to and from address is not or am I missing something?